rootston/seat.c

```diff
@@ -77,6 +77,78 @@ static void handle_cursor_axis(struct wl_listener *listener, void *data) {
 	roots_cursor_handle_axis(cursor, event);
 }
 
+static void handle_swipe_begin(struct wl_listener *listener, void *data) {
+	struct roots_cursor *cursor =
+		wl_container_of(listener, cursor, swipe_begin);
+	struct wlr_pointer_gestures_v1 *gestures = cursor->seat->input->server->desktop->pointer_gestures;
+	struct wlr_event_pointer_swipe_begin *event = data;
+	struct wl_resource *focus = roots_seat_get_focus(cursor->seat)->wlr_surface->resource;
+	struct wlr_pointer_gesture_swipe_v1 *swipe;
+	wl_list_for_each(swipe, &gestures->swipes, link) {
```

Comment:
If two clients create a `zwp_pointer_gesture_swipe_v1` object, both will receive swipe events regardless of which one is focused. Additionally, the one that isn't focused will receive events with an unknown surface (because it belongs to a different client).